### PR TITLE
bepasty: 0.5.0 -> 1.0.0

### PR DESCRIPTION
**Pull request by pinpox**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/issues/116326

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by dotlambda — CHANGES_REQUESTED** [review]
You need to update `sha256`.

**Comment by pinpox** [comment]
@dotlambda can you help me out? how do I get the sha256 for pypi references?

**Comment by SuperSandro2000** [comment]
> @dotlambda can you help me out? how do I get the sha256 for pypi references?

use nix-update or invalidate it and put the one in nix-build expected.

**Comment by dotlambda** [comment]
> use nix-update or invalidate it and put the one in nix-build expected.

Invalidating meaning replacing e.g. a leading 1 with a 0.
For PyPI, you get also get it from https://pypi.org/project/bepasty/#copy-hash-modal-9b2695cc-eba4-41ec-9369-d9bf4bdaaeda.

**Comment by pinpox** [comment]
> > use nix-update or invalidate it and put the one in nix-build expected.
> 
> Invalidating meaning replacing e.g. a leading 1 with a 0.
> For PyPI, you get also get it from https://pypi.org/project/bepasty/#copy-hash-modal-9b2695cc-eba4-41ec-9369-d9bf4bdaaeda.

I say the hashes shown on pypi.org, but they have they have the wrong length:
Old hash: 1y3smw9620w2ia4zfsl2svb9j7mkfgc8z1bzjffyk1w5vryhwikh
pypy : ae06ffc4562f678e09506c3e4587c955f83ab2d13fcbf62363a5d2c0edf64247

EDIT: I updated the hash with `nix-update` as @SuperSandro2000  suggested, shoud be right now.

**Review by dotlambda — CHANGES_REQUESTED** [review]
```
  ERROR: Could not find a version that satisfies the requirement XStatic-asciinema-player (from bepasty)
  ERROR: No matching distribution found for XStatic-asciinema-player
```
That library is not in Nixpkgs yet, so you need to add it.

**Comment by pinpox** [comment]
@dotlambda I copied one of the existing pypi modules and adapted it. Not very experienced with python packaging, hope I got that right. There where two libs missing.

**Review by dotlambda — CHANGES_REQUESTED** [review]
Since Font Awesome is already packaged in Nixpkgs, we should make use of https://xstatic.readthedocs.io/en/latest/packaging.html#notes-for-linux-or-other-os-package-maintainers for it.

We could also add a `checkPhase` that runs a little Python script:
```py
from xstatic.main import XStatic
__import__('xstatic.pkg', fromlist=['font-awesome'])
```

**Comment by dotlambda** [comment]
Btw, you need to have exactly one commit per package.

File: pkgs/development/python-modules/xstatic-asciinema-player/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+buildPythonPackage rec {
+  pname = "xstatic-asciinema-player";
+  version = "2.6.1.1";
+
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "c80e960b4ebb4adf360e6e92b5a08a756ad106198d7a6b307ad20ef22a1d7dcc";
+  };
```
Comment (makefu) [inline pkgs/development/python-modules/xstatic-asciinema-player/default.nix:11]:
```patch
diff --git a/pkgs/development/python-modules/xstatic-asciinema-player/default.nix b/pkgs/development/python-modules/xstatic-asciinema-player/default.nix
index e281f67387a..7605a2aabf5 100644
--- a/pkgs/development/python-modules/xstatic-asciinema-player/default.nix
+++ b/pkgs/development/python-modules/xstatic-asciinema-player/default.nix
@@ -6,7 +6,8 @@ buildPythonPackage rec {
 
 
   src = fetchPypi {
-    inherit pname version;
+    pname = "XStatic-asciinema-player";
+    inherit version;
     sha256 = "c80e960b4ebb4adf360e6e92b5a08a756ad106198d7a6b307ad20ef22a1d7dcc";
   };
```

File: pkgs/tools/misc/bepasty/default.nix [hunk 2]
```diff
@@ -1,35 +1,37 @@
...
-      });
+    packageOverrides = self: super: {
+      xstatic-bootstrap = super.xstatic-bootstrap.overridePythonAttrs
+        (oldAttrs: rec {
+          version = "3.3.7.1";
+          src = oldAttrs.src.override {
+            inherit version;
+            sha256 = "0cgihyjb9rg6r2ddpzbjm31y0901vyc8m9h3v0zrhxydx1w9x50c";
+          };
+        });
```
Comment (makefu) [inline pkgs/tools/misc/bepasty/default.nix:14]: The override must be deleted. bepasty uses now boostrap4 (which is what we have in the repo)

**Review by makefu — CHANGES_REQUESTED** [review]
after these changes bepasty builds again, however i have not yet tested it

**Comment by makefu** [comment]
when running `bepasty-server`:
```
[nix-shell:~/.cache/nixpkgs-review/pr-116429-1/nixpkgs]$ /nix/store/f33bd2pzfdq1753zsn2f63hvmz1fg3q3-python3.8-bepasty-1.0.0/bin/bepasty-server --port 9999
Error processing line 3 of /nix/store/wcl7qma99g31s3c78zvxlqcfkxpavj9d-python3.8-XStatic-1.0.2/lib/python3.8/site-packages/XStatic-1.0.2-py3.8-nspkg.pth:

  Traceback (most recent call last):
    File "/nix/store/66fbv9mmx1j4hrn9y06kcp73c3yb196r-python3-3.8.9/lib/python3.8/site.py", line 169, in addpackage
      exec(line)
    File "<string>", line 1, in <module>
    File "<frozen importlib._bootstrap>", line 553, in module_from_spec
  AttributeError: 'NoneType' object has no attribute 'loader'

Remainder of file ignored
 * Starting bepasty server...
 * Serving Flask app "bepasty.app" (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on http://127.0.0.1:9999/ (Press CTRL+C to quit)
127.0.0.1 - - [16/Jun/2021 22:52:26] "GET / HTTP/1.1" 200 -
```
there are some errors regarding xstatic, however it may not be an issue here? the page looks okay when loading it:
image

File: pkgs/development/python-modules/xstatic-font-awesome/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{ lib, buildPythonPackage, fetchPypi }:
+
+buildPythonPackage rec {
+  pname = "XStatic-Font-Awesome";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/xstatic-font-awesome/default.nix:4]:
```suggestion
  pname = "xstatic-font-awesome";
```
attr and pname should match most of the time.

File: pkgs/development/python-modules/xstatic-font-awesome/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  version = "4.7.0.0";
+
+  src = fetchPypi {
+    inherit pname version;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/xstatic-font-awesome/default.nix:8]:
```suggestion
    pname = "XStatic-Font-Awesome";
    inherit version;
```

**Comment by Lassulus** [comment]
So what's up here?

**Comment by makefu** [comment]
for me it looks good, i guess we can merge?

**Comment by Lassulus** [comment]
Result of `nixpkgs-review pr 116429` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>bepasty</li>
    <li>python38Packages.xstatic-asciinema-player</li>
    <li>python39Packages.xstatic-asciinema-player</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.xstatic-font-awesome</li>
    <li>python39Packages.xstatic-font-awesome</li>
  </ul>
</details>

**Comment by Lassulus** [comment]
does not look very good :D

**Comment by pinpox** [comment]
Sorry, I don't currently have the time to continue on this PR. Anybody is welcome to pick it up

**Pull request closed without merging**